### PR TITLE
Fix memory leak in UnitTestLinearSystem

### PR DESCRIPTION
**Pull request by alanw0**
Also clean up kokkos includes to stop a bunch of
warnings that were happening due to including non-public kokkos headers.

### CODE REVIEW

File: unit_tests/ngp_kernels/UnitTestNgpKernel.C [hunk 1]
```diff
@@ -112,4 +112,5 @@ TEST_F(Hex8MeshWithNSOFields, NGPKernelExecute)
   EXPECT_EQ(1u, assembleElemSolverAlg->activeKernels_.size());
 
   helperObjs.execute();
+std::cout<<"end of test comm: "<<bulk->parallel()<<std::endl;
```
Comment (psakievich) [inline unit_tests/ngp_kernels/UnitTestNgpKernel.C]: Is this a debugging statement, or a permanent addition?

File: unit_tests/UnitTestUtils.h [hunk 2]
```diff
@@ -91,6 +91,8 @@ class Hex8Mesh : public ::testing::Test
       coordField(nullptr),
       exactLaplacian(0.0)
   {
+std::cout<<"comm: "<<comm<<std::endl;
+std::cout<<"MPI_COMM_WORLD: "<<MPI_COMM_WORLD<<std::endl;
```
Comment (psakievich) [inline unit_tests/UnitTestUtils.h]: Left over debugging statements?

File: unit_tests/UnitTestLinearSystem.h [hunk 3]
```diff
@@ -262,6 +262,8 @@ class TestLinearSystem : public sierra::nalu::LinearSystem
     return newDeviceCoeffApplier;
   }
 
+  bool owns_coeff_applier() override { return false; }
```
Comment (psakievich) [inline unit_tests/UnitTestLinearSystem.h:265]: So was this missing implementation the source of the leak?

**Review by psakievich — APPROVED** [review]
LGTM

File: unit_tests/UnitTestLinearSystem.h (hunk 3 above)
Comment (alanw0) [reply unit_tests/UnitTestLinearSystem.h:265]: Yes. There is code elsewhere that frees the coeff-applier or not, based on whether it is owned by a LinearSystem object. This unit-test one is not owned.

**Review by PaulMullowney — APPROVED** [review]

**Pull request merged**